.github/workflows/azure_deploy_infra.yaml

```
@@ -0,0 +1,48 @@
+        default: "test"
+  push:
+    branches:
+      - main # Cambia il nome del branch se necessario
+
+jobs:
+  terraform:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```

actionlint

the runner of "actions/checkout@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)